lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj

```diff
@@ -95,12 +95,57 @@ public class Idl implements Closeable {
   String namespace;
   Map<String,Schema> names = new LinkedHashMap<String,Schema>();
 
-  private static final ThreadLocal<String> DOC = new ThreadLocal<String>();
-  static void setDoc(String doc) { DOC.set(doc.trim()); }
+  private static class DocComment {
+      private String text;
+      private int line;
+      private int column;
+      DocComment(Token token) {
+          // The token is everything after the initial '/**', including all whitespace and the ending '*/'
+          int tokenLength = token.image.length();
+          this.text = token.image.substring(0, tokenLength-2)./*replaceAll("[ \\t]*(\\r|\\r\\n|\\n)[ \\t]*\\*?[ \\t]*", "$1").*/trim();
```

Comment:
The commented out bit is the answer to Martins [comment in PR #1407](https://github.com/apache/avro/pull/1407#discussion_r754204005):